system/jlib/jmutex.hpp

```diff
@@ -1002,22 +1002,37 @@ template <typename X>
 class Singleton
 {
 public:
-    template <typename FUNC> X * query(FUNC factory) { return querySingleton(singleton, cs, factory); }
+    template <typename FUNC> X * query(FUNC factory)
+    {
```

Comment:
@ghalliday - considered adding a querySingleton() variant (that took an initializer atomic too), but I don't think it's worth it, or worth changing the existing direct uses of querySingleton (all in deftype.cpp)